### PR TITLE
[DOCS-1640] OAuth2 Introspection plugin add datatypes per schema

### PR DESCRIPTION
**Pull request by tlblessing**
https://konghq.atlassian.net/browse/DOCS-1640 part of epic https://konghq.atlassian.net/browse/DOCS-1396.

Schema links:

- https://github.com/Kong/kong-plugin-enterprise-oauth2-introspection/blob/master/kong/plugins/oauth2-introspection/schema.lua
- https://github.com/Kong/kong/blob/master/kong/db/schema/typedefs.lua (URL)

Direct review link:

https://deploy-preview-2718--kongdocs.netlify.app/hub/kong-inc/oauth2-introspection/

### CODE REVIEW

File: app/_hub/kong-inc/oauth2-introspection/index.md [hunk 1]
```diff
@@ -46,88 +47,101 @@ params:
...
       default:
       value_in_examples:
+      datatype: string
       description: |
-        The token_type_hint value to associate to introspection requests
+        The `token_type_hint` value to associate to introspection requests.
     - name: ttl
       required: false
-      default: 60
+      default: 30
```
Comment (murillopaula) [inline app/_hub/kong-inc/oauth2-introspection/index.md:69]: nice catch

File: app/_hub/kong-inc/oauth2-introspection/index.md [hunk 2]
```diff
@@ -46,88 +47,101 @@ params:
...
+        An optional value in milliseconds that defines how long an idle connection lives before being closed.
     - name: anonymous
       required: false
       default:
       value_in_examples:
+      datatype: string
       description: |
         An optional string (consumer uuid) value to use as an "anonymous" consumer if authentication fails. If empty (default), the request will fail with an authentication failure 4xx.
     - name: run_on_preflight
       required: false
```
Comment (murillopaula) [inline app/_hub/kong-inc/oauth2-introspection/index.md:103]:
```suggestion
      required: true
```

File: app/_hub/kong-inc/oauth2-introspection/index.md [hunk 3]
```diff
@@ -46,88 +47,101 @@ params:
...
-        `X-Request-Path` and `X-Request-Http-Method` will be inserted in the
-        introspect request
+        `X-Request-Path` and `X-Request-Http-Method` will be inserted into the
+        introspect request.
     - name: custom_introspection_headers
-      required: false
+      required: true
       default:
       value_in_examples:
+      datatype: map of string values
```
Comment (murillopaula) [inline app/_hub/kong-inc/oauth2-introspection/index.md:133]:
```suggestion
      datatype: map of string keys and string values
```

Can we add examples too? If I'm an user and see this datatype I'd love some examples at least in JSON.

File: app/_hub/kong-inc/oauth2-introspection/index.md (hunk 2 above)
Comment (murillopaula) [reply app/_hub/kong-inc/oauth2-introspection/index.md:103]: Just wanted to point out that it's a required field. Feel free to ignore this suggestion if you don't want to scope creep.

File: app/_hub/kong-inc/oauth2-introspection/index.md (hunk 3 above)
Comment (tlblessing) [reply app/_hub/kong-inc/oauth2-introspection/index.md:133]: Of course we can add them, but can you please provide them? So this request is scope creep beyond adding datatypes...

File: app/_hub/kong-inc/oauth2-introspection/index.md (hunk 2 above)
Comment (tlblessing) [reply app/_hub/kong-inc/oauth2-introspection/index.md:103]: This isn't one that had its schema explicitly updated for required booleans:

  { run_on_preflight = { type = "boolean", default = true } },

**Review by murillopaula — APPROVED** [review]

**Review by GayleNeumann — APPROVED** [review]
LGTM!

**Pull request merged**